code/src/components/TaskList.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable max-len */
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { tasks } from 'reducers/tasks'
+import { Checkbox } from './styles/Checkbox';
+import { IconButton } from './styles/IconButton';
+import { TaskListWrapper, SingleTaskWrapper } from './TaskListStyles';
+// for some very strange reason, I could only import these styles if i put them in the same folder as the TaskLis-component.
+
+export const TaskList = () => {
+  // FUNCTIONALITY
+  const taskList = useSelector((store) => store.tasks.items)
+  const dispatch = useDispatch()
+  const onTaskToggle = (taskId) => {
+    dispatch(tasks.actions.toggleTask({ taskId }));
```

Comment:
I'm not a hundred percent sure but I don't think you have to do object destructuring here, I think "toggleTask(taskId)" would be enough.